qutip/tests/test_superop_reps.py

```diff
@@ -148,204 +161,194 @@ def test_NeglectSmallKraus(self):
-
-        # Check that a linear combination of bipartitie unitaries is CPTP and HP.
-        S = (
-            to_super(tensor(sigmax(), identity(2))) + to_super(tensor(identity(2), sigmay()))
-        ) / 2
-        case(S, True, True, True)
-
-        # The partial transpose map, whose Choi matrix is SWAP, is TP
-        # and HP but not CP (one negative eigenvalue).
-        W = Qobj(swap(), type='super', superrep='choi')
-        case(W, True, False, True)
-
-        # Subnormalized maps (representing erasure channels, for instance)
-        # can be CP but not TP.
-        subnorm_map = Qobj(identity(4) * 0.9, type='super', superrep='super')
-        case(subnorm_map, True, True, False)
-
-        # Check that things which aren't even operators aren't identified as
-        # CPTP.
-        case(basis(2), False, False, False)
-
-    def test_choi_tr(self):
+        hp = qobj.ishp
+        cp = qobj.iscp
+        tp = qobj.istp
+        cptp = qobj.iscptp
+        shouldcptp = shouldcp and shouldtp
+
+        assert hp == shouldhp
+        assert cp == shouldcp
+        assert tp == shouldtp
+        assert cptp == shouldcptp
+
+    def test_choi_tr(self, dimension):
         """
         Superoperator: Trace returned by to_choi matches docstring.
         """
-        for dims in range(2, 5):
-            assert_(abs(to_choi(identity(dims)).tr() - dims) <= tol)
+        assert abs(to_choi(identity(dimension)).tr() - dimension) <= tol
+
 
-    def test_stinespring_cp(self, thresh=1e-10):
+    def test_stinespring_cp(self, dimension):
         """
         Stinespring: A and B match for CP maps.
         """
-        def case(map):
-            A, B = to_stinespring(map)
-            assert_(norm((A - B).full()) < thresh)
+        superop = rand_super_bcsz(dimension)
+        A, B = to_stinespring(superop)
 
-        for idx in range(4):
-            case(rand_super_bcsz(7))
+        assert norm(A - B) < tol
 
-    def test_stinespring_agrees(self, thresh=1e-10):
+    @pytest.mark.repeat(3)
+    def test_stinespring_agrees(self, dimension):
         """
         Stinespring: Partial Tr over pair agrees w/ supermatrix.
         """
-        def case(map, state):
-            S = to_super(map)
-            A, B = to_stinespring(map)
 
-            q1 = vector_to_operator(
-                S * operator_to_vector(state)
-            )
-            # FIXME: problem if Kraus index is implicitly
-            #        ptraced!
-            q2 = (A * state * B.dag()).ptrace((0,))
+        map = rand_super_bcsz(dimension)
+        state = rand_dm_ginibre(dimension)
 
-            assert_((q1 - q2).norm('tr') <= thresh)
+        S = to_super(map)
+        A, B = to_stinespring(map)
 
-        for idx in range(4):
-            case(rand_super_bcsz(2), rand_dm_ginibre(2))
+        q1 = vector_to_operator(
+            S * operator_to_vector(state)
+        )
+        # FIXME: problem if Kraus index is implicitly
+        #        ptraced!
```

Comment:
Could you explain the FIXME a bit more? Was this an existing issue? Can we just fix it?

Reply:
Yes, this was an existing issue.   I will check if the issue persists.

Reply:
I re-read Watrous' book's chapter on channel representation and I still can not pinpoint exactly what this refers to. I am sure I am missing something. Maybe @quantshah can help me here.